### PR TITLE
NFC first draft

### PR DESCRIPTION
**Pull request by markusbug**

### CODE REVIEW

File: apps/daimo-mobile/src/App.tsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ import { RpcProvider } from "./logic/trpc";
 import { useSyncChain } from "./sync/sync";
 import { HomeStackNav } from "./view/HomeStack";
 import { useInitNavLinks } from "./view/shared/nav";
+import { useInitNFC } from "./logic/nfcListen";
+
 
 export default function App() {
```
Comment (dcposch) [inline apps/daimo-mobile/src/App.tsx:13]: you can fix all the lint via VSCode > click Install Recommended in the extensions tab > save the file (should auto-format) > Cmd+Shift+P Organize Imports (fixes imports)

File: apps/daimo-mobile/src/logic/nfcListen.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import { MAX_NONCE_ID_SIZE_BITS } from "@daimo/userop";
+import { Hex } from "viem";
+import { generatePrivateKey } from "viem/accounts";
+import { useEffect } from "react";
+import { useNavigation } from "@react-navigation/native";
+import type { NativeStackNavigationProp } from "@react-navigation/native-stack";
+import { HomeStackParamList } from "../view/shared/nav";
+import { Recipient, getRecipient } from "../sync/recipients";
+
+NfcManager.start();
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/nfcListen.ts:11]: put this in a `useEffect` in `useInitNdef`

File: apps/daimo-mobile/src/logic/nfcListen.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+
+async function readNdef(nav: NativeStackNavigationProp<HomeStackParamList>) {
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/nfcListen.ts:14]: i'd add a short docstring on this function and on the exported `useInitNfc` that summarize what they do

File: apps/daimo-mobile/src/logic/nfcListen.ts [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    try {
+      // register for the NFC tag with NDEF in it
+      try {
+        await NfcManager.requestTechnology(NfcTech.IsoDep);
+      } catch (ex) {
+      }
+      // the resolved tag object will contain `ndefMessage` property
+      const tag = await NfcManager.getTag();
+      var tesp = await NfcManager.isoDepHandler.transceive([0x00, 0xA4, 0x04, 0x00, 0x07, 0xD2, 0x76, 0x00, 0x00, 0x85, 0x01, 0x01, 0x00]);
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/nfcListen.ts:23]: add link to wherever you got the constant from `// (nfc doc url)`

might consider `hexToBytes("0x00A4040007D276000085010100")`

File: apps/daimo-mobile/src/logic/nfcListen.ts [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  
+      const realResp = await NfcManager.isoDepHandler.transceive([0x00, 0xCA, 0x9A, 0x2B, 0x00]);
+  
+      const utf8String = String.fromCharCode.apply(null, realResp);
+  
+      await intepretInput(utf8String, nav);
+
+      NfcManager.cancelTechnologyRequest();
+
+      await sleep(2000)
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/nfcListen.ts:33]: why 2 second sleep?

File: apps/daimo-mobile/src/logic/nfcListen.ts [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+
+async function intepretInput(jsonStr: string, nav: NativeStackNavigationProp<HomeStackParamList>) {
+    var inputObj = JSON.parse(jsonStr);
+
+    const recipient = await getRecipient(inputObj.recipient);
+    const dollars = parseFloat(inputObj.amount);
+    nav.navigate("Send", { recipient, dollars, requestId: generateRequestID() });
+}
+
+function generateRequestID() {
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/nfcListen.ts:52]: request ID will come from the requesting side

File: apps/daimo-mobile/src/logic/nfcListen.ts [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    } catch (ex) {
+      console.warn('Oops!', ex);
+      NfcManager.cancelTechnologyRequest();
+      await sleep(2000)
+      readNdef(nav);
+    }    
+}
+
+async function intepretInput(jsonStr: string, nav: NativeStackNavigationProp<HomeStackParamList>) {
+    var inputObj = JSON.parse(jsonStr);
```
Comment (dcposch) [inline apps/daimo-mobile/src/logic/nfcListen.ts:45]: check out `parseDaimoLink` / `handleDeepLink` in `nav.ts`

easiest might be to send as a URL

if JSON is better, i recommend `const req: DaimoLinkRequest = JSON.parse(jsonStr)`

(i'll later add zod validation but not necessary in this PR)

File: apps/daimo-mobile/src/logic/nfcListen.ts (hunk 5 above)
Comment (markusbug) [reply apps/daimo-mobile/src/logic/nfcListen.ts:33]: Used to have it in there for testing, removing now

**Comment by dcposch** [comment]
Merging into an `nfc` branch, will fix lint as a follow-up PR.

**Pull request merged**